docs/python/programming-guide.md

```diff
@@ -7,236 +7,386 @@ pagination_prev: quickstart
 
 import LocalPostgres from '/docs/partials/_local_postgres.mdx';
 
+This guide shows you how to use DBOS to build Python apps that are **resilient to any failure**.
 
-This tutorial shows you how to use DBOS durable execution to make your Python app **resilient to any failure.**
-First, without using DBOS, we'll build an app that records greetings to two different systems: Postgres and an online guestbook.
-Then, we'll add DBOS durable execution to the app in **just four lines of code**.
-Thanks to durable execution, the app will always write to both systems consistently, even if it is interrupted or restarted at any point.
-
-## 1. Setting Up Your App
+## 1. Setting Up Your Environment
 
 Create a folder for your app with a virtual environment, then enter the folder and activate the virtual environment.
 
 <Tabs groupId="operating-systems" className="small-tabs">
 <TabItem value="maclinux" label="macOS or Linux">
 ```shell
-python3 -m venv greeting-guestbook/.venv
-cd greeting-guestbook
+python3 -m venv dbos-starter/.venv
+cd dbos-starter
 source .venv/bin/activate
 ```
 </TabItem>
 <TabItem value="win-ps" label="Windows (PowerShell)">
 ```shell
-python3 -m venv greeting-guestbook/.venv
-cd greeting-guestbook
+python3 -m venv dbos-starter/.venv
+cd dbos-starter
 .venv\Scripts\activate.ps1
 ```
 </TabItem>
 <TabItem value="win-cmd" label="Windows (cmd)">
 ```shell
-python3 -m venv greeting-guestbook/.venv
-cd greeting-guestbook
+python3 -m venv dbos-starter/.venv
+cd dbos-starter
 .venv\Scripts\activate.bat
 ```
 </TabItem>
 </Tabs>
 
-Then, install and initialize DBOS:
+Then, install DBOS and create a DBOS configuration file:
 ```shell
 pip install dbos
-dbos init
+dbos init --config
 ```
 
-Now, let's use FastAPI to write a simple app that greets our friends.
-Every time the app receives a greeting, it performs two steps:
+## 2. Workflows and Steps
 
-1. Sign an online guestbook with the greeting.
-2. Record the greeting in the database.
+Now, let's create the simplest interesting DBOS program.
+Create a `main.py` file and add this code to it:
 
-We deliberately **won't** use DBOS yet (except to fetch the database connection string) so we can show you how easy it is to add later.
+```python showLineNumbers title="main.py"
+from dbos import DBOS
 
-Copy the following code into `greeting_guestbook/main.py`, replacing its existing contents:
+DBOS()
 
-```python showLineNumbers title="greeting_guestbook/main.py"
-import logging
+@DBOS.step()
+def step_one():
+    print("Step one completed!")
 
-import requests
-from dbos import get_dbos_database_url
-from fastapi import FastAPI
-from sqlalchemy import create_engine
+@DBOS.step()
+def step_two():
+    print("Step two completed!")
 
-from .schema import dbos_hello
+@DBOS.workflow()
+def dbos_workflow():
+    step_one()
+    step_two()
 
-app = FastAPI()
-logging.basicConfig(level=logging.INFO)
+if __name__ == "__main__":
+    DBOS.launch()
```

Comment:
I think it'll help a lot if we describe when to call `DBOS()` (the initializer) and when to call `DBOS.launch()`. In this guide, after switching to FastAPI, the program no longer contains `DBOS.launch()`, which could be confusing. And historically, "when to init DBOS and when to use launch" is one of the most asked questions.